python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -137,7 +137,7 @@ def locations(self):
 
     def _setLFNnamePattern(self, lfn="", namePattern=""):
 
-        if self.defaultSE != "":
+        if hasattr(self, 'defaultSE') and self.defaultSE != "":
```

Comment:
Or even an XML where this parameter didn't exist as this will only be assigned to the object after the other XML attributes have been read in.